src/v/kafka/server/handlers/describe_log_dirs.cc

```diff
@@ -35,15 +35,27 @@ using partition_dir_set
   = absl::flat_hash_map<model::topic, std::vector<partition_data>>;
 
 static partition_data describe_partition(cluster::partition& p) {
-    auto local = describe_log_dirs_partition{
-      .partition_index = p.ntp().tp.partition(),
-      .partition_size = static_cast<int64_t>(p.size_bytes()),
-      .offset_lag = std::max(
-        p.high_watermark()() - p.dirty_offset()(), int64_t(0)),
-      .is_future_key = false,
-    };
-
-    return partition_data{.local = local};
+    auto result = partition_data{
+      .local = describe_log_dirs_partition{
+        .partition_index = p.ntp().tp.partition(),
+        .partition_size = static_cast<int64_t>(p.size_bytes()),
+        .offset_lag = std::max(
+          p.high_watermark()() - p.dirty_offset()(), int64_t(0)),
+        .is_future_key = false,
+      }};
+
+    auto cloud_space = p.cloud_log_size();
+    if (cloud_space.has_value()) {
```

Comment:
nit: `cloud_log_size` does not return an optional until the last commit

Reply:
nice catch

Reply:
oops, I've reordered commits